.github/workflows/regression-tests.yml

```diff
@@ -0,0 +1,472 @@
+name: Regression Tests
+
+on:
+  workflow_dispatch:
+    inputs:
+      hardhat-ref:
+        required: true
+        type: string
+        default: 'v-next'
+      edr-ref:
+        required: false
+        type: string
+        default: ''
+      repositories:
+        required: false
+        type: string
+        default: '["Elytro-eth/soul-wallet-contract", "foundry-rs/forge-std", "kalidao/keep", "mds1/multicall", "pancakeswap/infinity-core", "PaulRBerg/prb-math", "PaulRBerg/prb-proxy", "PaulRBerg/prb-test", "pcaversaccio/createx", "ProjectOpenSea/seaport", "sablier-labs/lockup", "sablier-labs/v2-periphery", "transmissions11/solmate", "Uniswap/UniswapX", "Vectorized/solady"]'
+      runners:
+        required: false
+        type: string
+        default: '["ubuntu-latest"]'
+      commands:
+        required: false
+        type: string
+        default: '["hardhat compile", "hardhat test solidity"]'
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-edr:
+    name: Build EDR
```

Comment:
Most of the edr bindings building job was copied from https://github.com/NomicFoundation/edr/blob/main/.github/workflows/edr-npm-release.yml.